fix(block-link): improve focus style to meet WCAG 2.2 AA

dancormier opened the pull request:
Resolves [STACKS-545](https://stackoverflow.atlassian.net/browse/STACKS-545)

> **Note**
> The above linked ticket shows updates as they relate to the `s-menu` component. The focusable elements shown within the `s-menu` component are `s-block-link`s, so this PR updates the `s-block-link` focus style.

dancormier committed:
fix(block-link): improve focus style to meet WCAG 2.2 AA

Resolves [STACKS-545](https://stackoverflow.atlassian.net/browse/STACKS-545)

CGuindon reviewed (commented):
<img width="159" alt="Screenshot 2024-01-30 at 7 40 31 AM" src="https://github.com/StackExchange/Stacks/assets/5436760/0cb5c0e1-992c-4b4f-9013-3586bd6b3272">

I'm assuming adding padding here won't do anything overall — devs would need to make sure the element has a container somewhere in the core code? 

Is there a better way to show an example of 'no container' that would show at least 2px padding from the text?

dancormier commented:
@CGuindon That example is kinda misleading. Each link has markup like this:

```html
<a href="…" class="s-block-link px0">…</a>
```

Note the `px0`. That class overrides any left or right padding to be `0px`, so a dev would have to specifically apply an atomic class to override the padding in this case. If we ever take a good look at our docs pages, I think this example would be a good one to get rid of.

I found instances of this in two files ([1](https://github.com/StackEng/StackOverflow/blob/main/StackOverflow/Views/SubCommunities/Articles/Edit.cshtml#L276), [2](https://github.com/StackEng/StackOverflow/blob/main/StackOverflow/Views/TopBar/Inbox.cshtml#L36)), so I don't think it's too big of a deal and it's probably something that can be resolved pretty easily in Core by finessing the padding a little.

dancormier committed:
fix(block-link): change focus ring from `black-050` to `white`

Related https://github.com/StackExchange/Stacks/pull/1623

giamir reviewed (approved):
Code changes look good to me. Thanks @dancormier.

I  see that there is a small divergence on the focus ring radius between the mockup

![Screenshot 2024-01-30 at 17 30 35](https://github.com/StackExchange/Stacks/assets/6231616/ee7609dd-42a8-4431-987b-a31c5ad3defc)

and the final result

![Screenshot 2024-01-30 at 17 30 28](https://github.com/StackExchange/Stacks/assets/6231616/9d6ac743-3e12-49bc-bb30-191a03e5b8c9)

dancormier committed:
Add border radius sm to block-link focus style

Addresses https://github.com/StackExchange/Stacks/pull/1621#pullrequestreview-1851801407
